controllers/handlers/passt/passt.go

```diff
@@ -0,0 +1,349 @@
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstCNIImageEnvV)
+	}
+	return nil
+}
+
+// NetworkBinding creates an InterfaceBindingPlugin for passt network binding
+func NetworkBinding(sidecarImage string) kubevirtcorev1.InterfaceBindingPlugin {
+	return kubevirtcorev1.InterfaceBindingPlugin{
+		NetworkAttachmentDefinition: NetworkAttachmentDefinition,
+		SidecarImage:                sidecarImage,
+		Migration:                   &kubevirtcorev1.InterfaceBindingMigration{},
+		ComputeResourceOverhead: &kubevirtcorev1.ResourceRequirementsWithoutClaims{
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: passtResourceMemory,
+			},
+		},
+	}
+}
+
+// GetImage gets the passt image from environment variable
+func GetImage() string {
+	passtImageOnce.Do(func() {
+		passtImage = os.Getenv(hcoutil.PasstImageEnvV)
+	})
+	return passtImage
+}
+
+// NewPasstBindingCNISA creates a ServiceAccount for the passt binding CNI
+func NewPasstBindingCNISA(hc *hcov1beta1.HyperConverged) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "passt-binding-cni",
+			Namespace: hc.Namespace,
+			Labels:    hcoutil.GetLabels(hcoutil.HyperConvergedName, hcoutil.AppComponentNetwork),
+		},
+	}
+}
+
+// NewPasstBindingCNIDaemonSet creates a DaemonSet for the passt binding CNI
+func NewPasstBindingCNIDaemonSet(hc *hcov1beta1.HyperConverged, isOpenShift bool) *appsv1.DaemonSet {
+	maxUnavailable := intstr.FromString("10%")
+
+	hostpath := "/opt/cni/bin"
+	if isOpenShift {
+		hostpath = "/var/lib/cni/bin"
+	}
+
+	spec := appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "passt-binding-cni",
+			},
+		},
+		UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+			Type: appsv1.RollingUpdateDaemonSetStrategyType,
+			RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+				MaxUnavailable: &maxUnavailable,
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name": "passt-binding-cni",
+					"tier": "node",
+					"app":  "passt-binding-cni",
+				},
+				Annotations: map[string]string{
+					"description": "passt-binding-cni installs 'passt binding' CNI on cluster nodes",
+				},
+			},
+			Spec: corev1.PodSpec{
+				PriorityClassName:  "system-cluster-critical",
+				ServiceAccountName: "passt-binding-cni",
+				Containers: []corev1.Container{
+					{
+						Name:  "installer",
+						Image: os.Getenv(hcoutil.PasstCNIImageEnvV),
+						Command: []string{
+							"/bin/sh",
+							"-ce",
+						},
+						Args: []string{
+							`ls -la "/cni/kubevirt-passt-binding"
+cp -f "/cni/kubevirt-passt-binding" "/opt/cni/bin"
+echo "passt binding CNI plugin installation complete..sleep infinity"
+sleep 2147483647`,
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("10m"),
```

Comment:
Can be fone once, as a global variable. No point to re-parse constant value again and again.